src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -284,6 +284,13 @@ class PeerActor(
 
       case GetStatus =>
         sender() ! StatusResponse(Handshaked(initialStatus))
+
+      case PeerActor.StartBlockBroadcast(blockchain) =>
+        val blockBroadcastActor = context.actorOf(
```

Comment:
We currently don't limit the number of broadcast actors per connection, I think we should add a separate task to look into that. I'll push a commit that monitors the BlockBroadcastActor termination,  you were right about that